src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotAuthzServiceUsageAclsStep.java

```diff
@@ -49,9 +53,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
 
     // Remove the custodian's access to bill this project to read requester pays bucket data.
     // The underlying service provides retries so we do not need to retry this operation
+    List<String> principalsToRemove =
+        new ArrayList<>(
+            List.of(policyEmails.get(IamRole.STEWARD), policyEmails.get(IamRole.READER)));
+    if (!snapshot
+        .getSourceDataset()
+        .getProjectResource()
+        .getServiceAccount()
+        .equals(tdrServiceAccountEmail)) {
+      principalsToRemove.add(snapshot.getSourceDataset().getProjectResource().getServiceAccount());
+    }
```

Comment:
Is there a way to remove this code duplication? Maybe by moving it into `Dataset`, and/or by changing the resource service APIs to accept `Dataset`?